simhash/__init__.py

```diff
@@ -8,44 +8,59 @@
 import numbers
 import collections
 from itertools import groupby
+import numpy as np
 
 if sys.version_info[0] >= 3:
     basestring = str
     unicode = str
     long = int
+
+    def int_to_bytes(n, length):
+        return n.to_bytes(length, 'big')
+
+    def bytes_to_int(b):
+        return int.from_bytes(b, 'big')
 else:
     range = xrange
 
+    def int_to_bytes(n, length):
+        return '{:0{}x}'.format(n, length * 2).decode('hex')
+
+    def bytes_to_int(b):
+        return int(b.encode('hex'), 16)
 
 def _hashfunc(x):
-    return int(hashlib.md5(x).hexdigest(), 16)
+    return hashlib.md5(x).digest()
 
 
 class Simhash(object):
+    # Constants used in calculating simhash. Larger values will use more RAM.
+    large_weight_cutoff = 50
+    batch_size = 200
 
     def __init__(
-            self, value, f=64, reg=r'[\w\u4e00-\u9fcc]+', hashfunc=None, log=None
+            self, value, f=64, reg=r'[\w\u4e00-\u9fcc]+', hashfunc=_hashfunc, log=None
     ):
         """
-        `f` is the dimensions of fingerprints
+        `f` is the dimensions of fingerprints, in bits. Must be a multiple of 8.
```

Comment:
This adds a limitation here. I just hope no one is using an uncommon value of `f`, for example `12`.

Reply:
Yeah for sure. I think that could be supported in theory -- you'd just have to round up to the next whole byte length for the intermediate calculations, and then truncate the final value. I didn't mess with implementing and testing it since it's unclear to me that anyone's using non-standard values.